app/scripts/directives/createSecret.js

```diff
@@ -113,7 +113,7 @@ angular.module("openshiftConsole")
 
         var constructSecretObject = function(data, authType) {
           var secret = {
-            apiVersion: "v1",
+            apiVersion: APIService.toAPIVersion(secretsVersion),
```

Comment:
I suggest we also call `APIService.objectToResourceGroupVersion` before we call `DataService.create` below to make sure the versions always match.